data-prepper-expression/src/main/antlr/DataPrepperStatement.g4

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+grammar DataPrepperStatement;
+
+@header {
+    package org.opensearch.dataprepper.expression.antlr;
+}
+
+fragment
+Digit
+    : '0'
+    | NonZeroDigit
+    ;
+
+fragment
+NonZeroDigit
+    : [1-9]
+    ;
+
+Integer
+    : '0'
+    | NonZeroDigit Digit*
+    ;
+
+Float
+    : NonZeroDigit? Digit '.' Digit
+    | NonZeroDigit? Digit '.' Digit+ NonZeroDigit
+    | '.' Digit
+    | '.' Digit* NonZeroDigit
+    ;
+
+Boolean
+    : TRUE
+    | FALSE
+    ;
+
+fragment
+StringCharacters
+    : StringCharacter+
+    ;
+
+fragment
+StringCharacter
+    : ~["\\$]
+    | EscapeSequence
+    ;
+
+fragment
+EscapeSequence
+    : '\\' [btnfr"'\\$]
+    ;
+
+fragment
+JsonPointerEscapeSequence
+    : '\\' [btnfr"'\\/]
+    ;
+
+fragment
+JsonPointerStringCharacters
+    : JsonPointerStringCharacter+
+    ;
+
+fragment
+JsonPointerStringCharacter
+    : ~["\\]
+    | JsonPointerEscapeSequence
+    ;
+
+fragment
+JsonPointerCharacter
+    : [A-Za-z0-9_]
+    ;
+
+fragment
+JsonPointerCharacters
+    : JsonPointerCharacter+
+    ;
+
+JsonPointer
+    : DOUBLEQUOTE FORWARDSLASH JsonPointerStringCharacters? DOUBLEQUOTE
+    | FORWARDSLASH JsonPointerCharacters? (FORWARDSLASH JsonPointerCharacters)*
+    ;
+
+String
+    : DOUBLEQUOTE StringCharacters? DOUBLEQUOTE
+    ;
+
+fragment
+VariableNameLeadingCharacter
+    : [A-Za-z_]
+    ;
+
+fragment
+VariableNameCharacter
+    : VariableNameLeadingCharacter
+    | [0-9-]
+    ;
+
+fragment
+VariableNameCharacters
+    : VariableNameLeadingCharacter VariableNameCharacter*
+    ;
+
+VariableIdentifier
+    : '${' VariableNameCharacters '}'
+    ;
+
+statement
```

Comment:
If I understand this correctly, the tree will start with a statement. Could we start with an expression instead?